features/steps/jobs_steps.go

```diff
@@ -0,0 +1,154 @@
+	"github.com/ONSdigital/dp-search-reindex-api/models"
+	"github.com/ONSdigital/dp-search-reindex-api/service"
+	"github.com/ONSdigital/dp-search-reindex-api/service/mock"
+	"github.com/cucumber/godog"
+	"github.com/pkg/errors"
+	"github.com/rdumont/assistdog"
+	uuid "github.com/satori/go.uuid"
+	"github.com/stretchr/testify/assert"
+	"io/ioutil"
+	"net/http"
+	"strconv"
+	"strings"
+	"time"
+)
+
+type JobsFeature struct {
+	ErrorFeature   componenttest.ErrorFeature
+	svc            *service.Service
+	errorChan      chan error
+	Config         *config.Config
+	HTTPServer     *http.Server
+	ServiceRunning bool
+	ApiFeature     *componenttest.APIFeature
+	responseBody   []byte
+}
+
+func NewJobsFeature() (*JobsFeature, error) {
+	f := &JobsFeature{
+		HTTPServer:     &http.Server{},
+		errorChan:      make(chan error),
+		ServiceRunning: false,
+	}
+	svcErrors := make(chan error, 1)
+	cfg, err := config.Get()
+	if err != nil {
+		return nil, err
+	}
+	initFunctions := &mock.InitialiserMock{
+		DoGetHealthCheckFunc: f.DoGetHealthcheckOk,
+		DoGetHTTPServerFunc:  f.DoGetHTTPServer,
+	}
+	ctx := context.Background()
+	serviceList := service.NewServiceList(initFunctions)
+	f.svc, err = service.Run(ctx, cfg, serviceList, "1", "", "", svcErrors)
+	if err != nil {
+		return nil, err
+	}
+
+	return f, nil
+}
+
+func (f *JobsFeature) InitAPIFeature() *componenttest.APIFeature {
+	f.ApiFeature = componenttest.NewAPIFeature(f.InitialiseService)
+
+	return f.ApiFeature
+}
+
+func (f *JobsFeature) RegisterSteps(ctx *godog.ScenarioContext) {
+	ctx.Step(`^I would expect id, last_updated, and links to have this structure$`, f.iWouldExpectIdLast_updatedAndLinksToHaveThisStructure)
+	ctx.Step(`^the response should also contain the following values:$`, f.theResponseShouldAlsoContainTheFollowingValues)
+
+}
+func (f *JobsFeature) Reset() *JobsFeature {
+	return f
+}
+func (f *JobsFeature) Close() error {
+	if f.svc != nil && f.ServiceRunning {
+		f.svc.Close(context.Background())
+		f.ServiceRunning = false
+	}
+	return nil
+}
+func (f *JobsFeature) InitialiseService() (http.Handler, error) {
+	return f.HTTPServer.Handler, nil
+}
+func (f *JobsFeature) DoGetHTTPServer(bindAddr string, router http.Handler) service.HTTPServer {
+	f.HTTPServer.Addr = bindAddr
+	f.HTTPServer.Handler = router
+	return f.HTTPServer
+}
+
+func (f *JobsFeature) DoGetHealthcheckOk(cfg *config.Config, time string, commit string, version string) (service.HealthChecker, error) {
+	versionInfo, _ := healthcheck.NewVersionInfo(time, commit, version)
+	hc := healthcheck.New(versionInfo, cfg.HealthCheckCriticalTimeout, cfg.HealthCheckInterval)
+	return &hc, nil
+}
+
+func (f *JobsFeature) iWouldExpectIdLast_updatedAndLinksToHaveThisStructure(table *godog.Table) error {
+	f.responseBody, _ = ioutil.ReadAll(f.ApiFeature.HttpResponse.Body)
+	assist := assistdog.NewDefault()
+
+	expectedResult, err := assist.ParseMap(table)
+	if err != nil {
+		panic(err)
```

Comment:
Is this panic intentional or should it return the error?

Reply:
To be honest this code came from Armakuni so I don't know the answer. I can return the error if you think that would be better?

Reply:
I'm not sure either I've not see or worked with Armakuni yet or seen examples of how to write these tests so I cannot comment. If its taken from their example happy to leave as is.